### PR TITLE
DependentService refactoring

### PR DESCRIPTION
**Pull request by ytqsl**
## Description

Implements [Task 14495](https://dev.azure.com/qslvic/Health%20Gateway/_workitems/edit/14495): DependentService refactoring

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## UI Changes

None

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by rayking29 — DISMISSED** [review]

File: Apps/Common/src/Factories/RequestResultFactory.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    using System.Collections.Generic;
+    using System.Linq;
+    using HealthGateway.Common.Data.ErrorHandling;
+    using HealthGateway.Common.Data.ViewModels;
+    using HealthGateway.Common.ErrorHandling;
+
+    /// <summary>
+    /// Factory for  <see cref="RequestResult{T}"/> instances.
+    /// </summary>
+    public static class RequestResultFactory
```
Comment (BrianMaki) [inline Apps/Common/src/Factories/RequestResultFactory.cs:28]: Are there unit tests for this static factory?  Will this affect code coverage if there aren't any unit tests?

File: Apps/Common/src/Validations/VaccineStatusQueryValidator.cs [hunk 2]
```diff
@@ -17,7 +17,7 @@
 namespace HealthGateway.Common.Validations
 {
     using FluentValidation;
-    using HealthGateway.Common.Data.Validation;
```
Comment (BrianMaki) [inline Apps/Common/src/Validations/VaccineStatusQueryValidator.cs:20]: I think this class was introduced in a previous PR but can this method be unit tested directly or is it being tested indirectly through a service?

File: Apps/CommonData/src/Validations/DependantAgeValidator.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace HealthGateway.Common.Data.Validations
+{
+    using System;
+    using FluentValidation;
+
+    /// <summary>
+    /// Validates a dependent's age against a maximum value using date of birth.
+    /// </summary>
+    public class DependantAgeValidator : AbstractValidator<DateTime>
```
Comment (BrianMaki) [inline Apps/CommonData/src/Validations/DependantAgeValidator.cs:25]: Is there or can there be a unit test associated with this validator?

File: Apps/GatewayApi/src/Services/DependentService.cs [hunk 4]
```diff
@@ -81,103 +83,56 @@ public class DependentService : IDependentService
...
-                {
-                    ResultStatus = ResultType.ActionRequired,
-                    ResultError = ErrorTranslator.ActionRequired(ErrorMessages.InvalidServicesCard, ActionType.NoHdId),
-                };
+                    break;
             }
 
             // Insert Dependent to database
-            ResourceDelegate dependent = new()
+            var dependent = new ResourceDelegate()
```
Comment (BrianMaki) [inline Apps/GatewayApi/src/Services/DependentService.cs:118]: lines 118 and line 126.  Why was this changed?  Before you onboarded there were a lot changes made to avoid warnings.  When I try to make that change you have done on line 126, my Rider IDE will not allow me to do that.

File: Apps/GatewayApi/test/unit/Services.Test/DependentServiceTests.cs [hunk 5]
```diff
@@ -250,7 +250,7 @@ public void ValidateRemove()
             mockUserProfileDelegate.Setup(s => s.GetUserProfile(this.mockParentHdid))
                 .Returns(
                     new DbResult<UserProfile>
-                        { Payload = new UserProfile() });
+                    { Payload = new UserProfile() });
```
Comment (BrianMaki) [inline Apps/GatewayApi/test/unit/Services.Test/DependentServiceTests.cs:253]: Lines 253 and 416. You've changed the formatting.  I believe the team has tried to standardize formatting so that when files are saved and then checked, there aren't a lot of changes.

**Review by BrianMaki — COMMENTED** [review]
See my comments.

File: Apps/Common/src/Factories/RequestResultFactory.cs (hunk 1 above)
Comment (ytqsl) [reply Apps/Common/src/Factories/RequestResultFactory.cs:28]: There is no specific unit tests to cover the factory methods, however they're covered in the services that use these methods. There's little value in testing these methods as they're merely construction shortcuts without logic worth testing,

File: Apps/GatewayApi/src/Services/DependentService.cs (hunk 4 above)
Comment (ytqsl) [reply Apps/GatewayApi/src/Services/DependentService.cs:118]: I find the C# 9.0 constructor invocation pattern less readable and less flexible when following OO/SOLID principles (https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new for some caveats with it).
I don't use Rider yet so I don't know why it wouldn't let you change the syntax to standard 20 years old C#.

File: Apps/GatewayApi/test/unit/Services.Test/DependentServiceTests.cs (hunk 5 above)
Comment (ytqsl) [reply Apps/GatewayApi/test/unit/Services.Test/DependentServiceTests.cs]: fixed

File: Apps/Common/src/Validations/VaccineStatusQueryValidator.cs (hunk 2 above)
Comment (ytqsl) [reply Apps/Common/src/Validations/VaccineStatusQueryValidator.cs:20]: Added unit  tests

File: Apps/CommonData/src/Validations/DependantAgeValidator.cs (hunk 3 above)
Comment (ytqsl) [reply Apps/CommonData/src/Validations/DependantAgeValidator.cs:25]: added unit tests

File: Apps/GatewayApi/src/Services/DependentService.cs (hunk 4 above)
Comment (BrianMaki) [reply Apps/GatewayApi/src/Services/DependentService.cs:118]: If you like look at our code base, that seems to be the standard. I recall a task earlier to fix the code base to ensure new() was being used correctly.  I will approve but you may want to discuss with @sslaws

File: Apps/Common/src/Factories/RequestResultFactory.cs (hunk 1 above)
Comment (BrianMaki) [reply Apps/Common/src/Factories/RequestResultFactory.cs:28]: fine

**Review by BrianMaki — APPROVED** [review]

**Review by rayking29 — APPROVED** [review]
